doc/mesh-adapter-api.txt

```diff
@@ -0,0 +1,137 @@
+BlueZ D-Bus Mesh Adapter API description
+********************************
+
+Mesh Network Hierarchy
+======================
+Service		org.bluez.mesh
+Interface	org.bluez.mesh.AdapterManager1
+Object path	/org/bluez/mesh/adapter
+
+Methods:
+	uint64 token Add(string io_type, dict config)
+
+	    This is method that an application has to call to
+	    add a new mesh adapter with an IO type.
+
+	    The config parameter is a dictionary with configuration
+	    options specific for the chosen IO type. In case of "tcpserver",
+	    mesh adapter with given UUID and NetworkKey can added only once.
+	    When adapter with UUID and NetworkKey is already exists then
+	    AlreadyExists error will be raised. When one of required
+	    configuration options is missing then InvalidArguments
+	    error will be raised.
+
+        string io_type
+
+            This parameter choose which IO type will be used
+            to communicate with an adapter.
+
+            The defined values are: "tcpserver", "silvair"
+
+        dict config
+
+            Options available for the "tcpserver" IO:
+
+                int Port
+
+                    Port for listening for incoming connections.
+                    This field is mandatory.
+
+                array{byte}[16] UUID
+
+                    uGateway UUID in 16-byte array format.
```

Comment:
This doesn't need to be "uGateway" - change the wording to mention that this data is used to derive TLS keys, and describe how.